Releasing.md

```diff
@@ -170,12 +170,13 @@ For the body of the post, just copy this checklist and again replace all occurre
 <p>o Ensure that the bundle files are updated to include any changes to the release branch by running <code>npm run bundle</code> in gutenberg-mobile release branch and committing any changes. </p>
 <!-- /wp:paragraph -->
 
+
 <!-- wp:paragraph -->
-<p>o Merge the <strong>gutenberg-mobile</strong> PR to <code>trunk</code>. Use "Create a merge commit" option when merging, otherwise there could be conflicts between <code>trunk</code> and release branch in the next release. WARNING: Do not merge the <strong>gutenberg</strong> PR into <code>trunk</code> at this point.</p>
+<p>o <a href="https://github.com/wordpress-mobile/gutenberg-mobile/releases/new?tag=vX.XX.X&amp;target=release/X.XX.X&amp;title=Release%20X.XX.X">Create a new gutenberg-mobile GitHub Release</a>. Include a list of changes in the Release description.</p>
 <!-- /wp:paragraph -->
 
 <!-- wp:paragraph -->
-<p>o <a href="https://github.com/wordpress-mobile/gutenberg-mobile/releases/new?tag=vX.XX.X&amp;target=trunk&amp;title=Release%20X.XX.X">Create a new gutenberg-mobile GitHub Release</a>. Include a list of changes in the Release description.</p>
+<p>o Merge the <strong>gutenberg-mobile</strong> PR to <code>trunk</code>. Use "Create a merge commit" option when merging, otherwise there could be conflicts between <code>trunk</code> and release branch in the next release. WARNING: Do not merge the <strong>gutenberg</strong> PR into <code>trunk</code> at this point.</p>
```

Comment:
We can probably drop the:

> WARNING: Do not merge the <strong>gutenberg</strong> PR into <code>trunk</code> at this point.

Also, I think we need to address how to handle merge conflicts here like [we currently do for merging `trunk` back to develop](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blame/develop/Releasing.md#L224) (at a minimum, the Gutenberg hash is likely going to be a conflict).

if we want. I'd actually be inclined to reorder these steps a bit from the current:

1. merge gb-mobile to "trunk" (our old trunk branch)
2. Create gb-mobile release from "trunk"
3. merge gb PR
4. ...

to now be:

1. Create gb-mobile release off of release branch (you've already made the change to move this to be first)
2. Do the same merge domino we know and love:
    1. **Merge GB PR.** It feels more natural to me to do this merge before the gb-mobile merge.
    2. **Merge gb-mobile PR.** Like I mentioned earlier, we may need to also include instructions for creating a separate merge branch here to deal with merge conflicts. This might be another factor in favor of not doing that post-release merge and instead only adding stuff to the release branch off of trunk like they do in the gb repo.
    3. Merge main app PRs
 
Obviously, this is assuming that we will continue with our post-release merge of the release branches back to `trunk`, which is currently being discussed) I think it makes sense to assume we'll stick with the current approach of merging release branches back to `trunk` in this PR though; if we reach a different decision we can address that in a separate PR.

Reply:
> this is assuming that we will continue with our post-release merge of the release branches back to trunk

I'm also assuming that we continue to do this for the foreseeable future.

Reply:
> we may need to also include instructions for creating a separate merge branch here to deal with merge conflicts.

I'm not sure I follow the need for a third branch just to handle merge conflicts. Can't the conflicts be resolved in the branch being merged into trunk?

Reply:
Yeah, the only reason I'd argue about solving the conflicts in a separate branch, is in case we'd like to branch off from the release branch for cutting a beta/hotfixes. In any case, we could still do this by using the release tags, so no need to create a third branch.